crates/oxc_linter/src/rules/eslint/dot_notation.rs

```diff
@@ -0,0 +1,434 @@
+use crate::{context::LintContext, rule::Rule, AstNode};
+use lazy_regex::Regex;
+use oxc_ast::ast::{ComputedMemberExpression, Expression, StaticMemberExpression};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use oxc_syntax::number::NumberBase;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::ops::Deref;
+
+fn use_dot_notation_diagnostic(span: Span, identifier: &str, key: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("[\"{key}\"] is better written in dot notation."))
+        .with_help(format!("Replace {identifier}[\"{key}\"] with {identifier}.{key}"))
+        .with_label(span)
+}
+
+fn use_brackets_diagnostic(span: Span, identifier: &str, key: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(".{key} is a syntax error."))
+        .with_help(format!("Replace {identifier}.{key} with {identifier}[\"{key}\"]"))
+        .with_label(span)
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[schemars(rename_all = "camelCase")]
+pub struct DotNotationConfig {
+    allow_keywords: bool,
+    allow_pattern: Option<String>,
```

Comment:
I would prefer using `allow_pattern: Option<Regex>` here, but I got the error
> "the trait `JsonSchema` is not implemented for `lazy_regex::Regex`"

Reply:
yeah, this is a known problem for the moment that I haven't gotten around to solving yet: https://github.com/oxc-project/oxc/issues/15203

You can do `schemars(skip)` for this line and just add a TODO comment, if you want to use Regex here.

Reply:
Okay, thanks for the hint. I will then keep it as it is for now.

Reply:
@pchorus I fixed this in the oxc-schemars crate, it should get pulled in by renovate automatically pretty soon. See also #15978. So if you want to go back to using Regex here, it should work fine after rebasing (assuming the PR to upgrade schemars has been merged).

Reply:
@connorshea Nice! Thank you for that fix. I rebased my branch and added a fixup commit turning the option into a `Regex`. Much nicer now. 👍🏻